bindings/sundials4py/sundials/generate.yaml

```
@@ -0,0 +1,154 @@
+# -----------------------------------------------------------------
+# Programmer(s): Cody J. Balos @ LLNL
+# -----------------------------------------------------------------
+# SUNDIALS Copyright Start
+# Copyright (c) 2002-2025, Lawrence Livermore National Security
+# and Southern Methodist University.
+# All rights reserved.
+#
+# See the top-level LICENSE and NOTICE files for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+# SUNDIALS Copyright End
+# -----------------------------------------------------------------
+# This YAML file is used to instruct the generate.py script on
+# how to generate some of the nanobind code needed for the
+# core module of sundials4py.
+# -----------------------------------------------------------------
+
+modules:
+  all:
+    fn_exclude_by_name__regex:
+    - "Free" # Free and destroy functions should not need to be called as all objects on the Python side are RAII objects
+    - "Destroy"
+    - "NewEmpty" # No need to call NewEmpty functions in Python
+    - "CopyOps" # No need to call CopyOps functions in Python
+    - "Space" # Space functions are deprecated, so dont expose them in Python
+    # Due to the need to convert between sys.argv and C argv, we need to do custom wrappers of these
+    - "SetOptions" 
+    macro_define_include_by_name__regex:
+    - "^SUN_"
+  sundials_adaptcontroller:
+    path: sundials/sundials_adaptcontroller_generated.hpp
+    headers:
+    - ../../include/sundials/sundials_adaptcontroller.h
+  sundials_adjointcheckpointscheme:
+    path: sundials/sundials_adjointcheckpointscheme_generated.hpp
+    headers:
+    - ../../include/sundials/sundials_adjointcheckpointscheme.h
+    fn_exclude_by_name__regex:
+    - "Set.*Fn" # nanobind cannot bind to functions which take a function pointer, so we do something custom
+    - "SetContent" # no need to set content from Python, so we don't interface it
+    - "GetContent" # content should not be accessible from Python, so we don't interface it
+  sundials_adjointstepper:
+    path: sundials/sundials_adjointstepper_generated.hpp
+    headers:
+    - ../../include/sundials/sundials_adjointstepper.h
+  sundials_context:
+    path: sundials/sundials_context_generated.hpp
+    headers:
+    - ../../include/sundials/sundials_context.h
+    fn_exclude_by_name__regex:
+    - "^SUNContext_PushErrHandler$"
```

Comment to different from functions we don't provide wrappers for 
```suggestion
    - "^SUNContext_PushErrHandler$" # nanobind cannot bind to functions which take functions points, so we do do something custom 
```